typesystem/serialization/json/vector.h

```diff
@@ -45,17 +45,47 @@ struct SerializeImpl<json::JSONStringifier<JSON_FORMAT>, std::vector<T>> {
   }
 };
 
-template <class JSON_FORMAT, typename TT, typename TA>
-struct DeserializeImpl<json::JSONParser<JSON_FORMAT>, std::vector<TT, TA>> {
-  static void DoDeserialize(json::JSONParser<JSON_FORMAT>& json_parser, std::vector<TT, TA>& destination) {
+template <class JSON_FORMAT, typename TA>
+struct SerializeImpl<json::JSONStringifier<JSON_FORMAT>, std::vector<bool, TA>> {
+  static void DoSerialize(json::JSONStringifier<JSON_FORMAT>& json_stringifier, const std::vector<bool, TA>& value) {
+    json_stringifier.Current().SetArray();
+    for (const auto& element : value) {
+      rapidjson::Value element_to_push;
+      const bool tmp = element;
+      json_stringifier.Inner(&element_to_push, tmp);
+      json_stringifier.Current().PushBack(std::move(element_to_push.Move()), json_stringifier.Allocator());
+    }
+  }
+};
+
+template <class JSON_FORMAT, typename T, typename TA>
+struct DeserializeImpl<json::JSONParser<JSON_FORMAT>, std::vector<T, TA>> {
+  static void DoDeserialize(json::JSONParser<JSON_FORMAT>& json_parser, std::vector<T, TA>& destination) {
     if (json_parser && json_parser.Current().IsArray()) {
       const size_t size = json_parser.Current().Size();
       destination.resize(size);
       for (rapidjson::SizeType i = 0; i < static_cast<rapidjson::SizeType>(size); ++i) {
         json_parser.Inner(&json_parser.Current()[i], destination[i], "[", static_cast<int>(i), "]");
       }
     } else if (!json::JSONPatchMode<JSON_FORMAT>::value || (json_parser && !json_parser.Current().IsArray())) {
-      throw JSONSchemaException("array", json_parser);  // LCOV_EXCL_LINE
+      CURRENT_THROW(JSONSchemaException("array", json_parser));  // LCOV_EXCL_LINE
+    }
+  }
+};
+
+template <class JSON_FORMAT, typename TA>
+struct DeserializeImpl<json::JSONParser<JSON_FORMAT>, std::vector<bool, TA>> {
+  static void DoDeserialize(json::JSONParser<JSON_FORMAT>& json_parser, std::vector<bool, TA>& destination) {
+    if (json_parser && json_parser.Current().IsArray()) {
+      const size_t size = json_parser.Current().Size();
+      destination.resize(size);
+      for (rapidjson::SizeType i = 0; i < static_cast<rapidjson::SizeType>(size); ++i) {
+        bool tmp;
+        json_parser.Inner(&json_parser.Current()[i], tmp, "[", static_cast<int>(i), "]");
+        destination[i] = tmp;
+      }
+    } else if (!json::JSONPatchMode<JSON_FORMAT>::value || (json_parser && !json_parser.Current().IsArray())) {
+      CURRENT_THROW(JSONSchemaException("array", json_parser));  // LCOV_EXCL_LINE
```

Comment:
What for?

Reply:
I don't know what for (maybe because the macro expands into more than one line), but you made the same change in this PR in another line.
<img width="992" alt="screen shot 2018-01-02 at 12 52 38 pm" src="https://user-images.githubusercontent.com/498274/34499639-dba3c2ba-efbb-11e7-866f-31f43a762d65.png">

Reply:
Most likely because the line was too long so I had to break it.